### PR TITLE
Add kubeletConfig to nodeBootstrapping

### PR DESCRIPTION
**Pull request by anujmaheshwari1**
Move KubeletConfig to NodeBootstrappingConfig

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -261,8 +261,8 @@ func validateAndSetLinuxNodeBootstrappingConfiguration(config *datamodel.NodeBoo
 	// If using kubelet config file, disable DynamicKubeletConfig feature gate and remove dynamic-config-dir
 	// we should only allow users to configure from API (20201101 and later)
 	if IsKubeletConfigFileEnabled(config.ContainerService, config.AgentPoolProfile, config.EnableKubeletConfigFile) {
-		if config.AgentPoolProfile.KubernetesConfig != nil && config.AgentPoolProfile.KubernetesConfig.KubeletConfig != nil {
```
Comment (xuto2) [inline pkg/agent/baker.go:264]: L275 should also be updated?

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -315,7 +315,7 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			if profile.KubernetesConfig == nil {
 				return ""
 			}
-			return GetKubeletConfigFileContent(profile.KubernetesConfig.KubeletConfig, profile.CustomKubeletConfig)
```
Comment (xuto2) [inline pkg/agent/baker.go:318]: L315-L317 is not needed, right?

File: pkg/agent/datamodel/types.go [hunk 3]
```diff
@@ -1427,6 +1427,7 @@ type NodeBootstrappingConfiguration struct {
 	KubeletClientTLSBootstrapToken *string
 	FIPSEnabled                    bool
 	HTTPProxyConfig                *HTTPProxyConfig
+	KubeletConfig                  map[string]string
```
Comment (xuto2) [inline pkg/agent/datamodel/types.go:1430]: can we remove KubeletConfig from type KubernetesConfig struct ? I think you will find some more places referencing this after doing that

File: pkg/agent/baker.go (hunk 1 above)
Comment (anujmaheshwari1) [reply pkg/agent/baker.go:264]: yeah, sorry I did not touch the windows functions. updated.

File: pkg/agent/datamodel/types.go [hunk 4]
```diff
@@ -1342,15 +1341,15 @@ func (k *KubernetesConfig) GetAzureCNIURLWindows(cloudSpecConfig *AzureEnvironme
 }
 
 // GetOrderedKubeletConfigStringForPowershell returns an ordered string of key/val pairs for Powershell script consumption
-func (k *KubernetesConfig) GetOrderedKubeletConfigStringForPowershell() string {
+func (config *NodeBootstrappingConfiguration) GetOrderedKubeletConfigStringForPowershell() string {
 	keys := []string{}
-	for key := range k.KubeletConfig {
+	for key := range config.KubeletConfig {
```
Comment (xuto2) [inline pkg/agent/datamodel/types.go:1346]: check if config is nil or not

**Comment by xuto2** [comment]
Be sure to update RP when this change is vendored in RP.

**Review by xuto2 — APPROVED** [review]

File: pkg/agent/datamodel/types.go [hunk 5]
```diff
@@ -1342,15 +1341,19 @@ func (k *KubernetesConfig) GetAzureCNIURLWindows(cloudSpecConfig *AzureEnvironme
...
+		return ""
+	}
+	
 	keys := []string{}
-	for key := range k.KubeletConfig {
+	for key := range config.KubeletConfig {
 		keys = append(keys, key)
 	}
 	sort.Strings(keys)
 	var buf bytes.Buffer
```
Comment (eriksywu) [inline pkg/agent/datamodel/types.go:1354]: nit: `strings.Builder` though underneath I'm sure it's just a byte buffer :P

**Review by eriksywu — APPROVED** [review]

File: pkg/agent/utils.go [hunk 6]
```diff
@@ -350,21 +350,21 @@ func getCustomDataFromJSON(jsonStr string) string {
 
 // GetOrderedKubeletConfigFlagString returns an ordered string of key/val pairs
 // copied from AKS-Engine and filter out flags that already translated to config file
-func GetOrderedKubeletConfigFlagString(k *datamodel.KubernetesConfig, cs *datamodel.ContainerService, profile *datamodel.AgentPoolProfile, kubeletConfigFileToggleEnabled bool) string {
-	if k.KubeletConfig == nil {
+func GetOrderedKubeletConfigFlagString(k map[string]string, cs *datamodel.ContainerService, profile *datamodel.AgentPoolProfile, kubeletConfigFileToggleEnabled bool) string {
```
Comment (eriksywu) [inline pkg/agent/utils.go:353]: nit: better naming

**Comment by mainred** [comment]
It looks good, and after we changed the bake_test.go and no generated data is changed.
We are moving config.AgentPoolProfile.KubernetesConfig.KubeletConfig to config.KubeletConfig, does this mean we are on the way to get rid of AgentPoolProfile from nodeBootstrapping or what's the plan? Thanks.

**Review by mainred — APPROVED** [review]

**Pull request merged**